File: Module II/13. Databases/EXAM - DB/01. Code first/Superheroes/Superheroes.Models/City.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Superheroes.Models
{
    public class City
    {
        private ICollection<Superhero> superheroes;

        public City()
        {
            this.superheroes = new HashSet<Superhero>();
        }

        public int Id { get; set; }

        [Index(IsUnique = true)]
        [MaxLength(30), MinLength(2)]
        [Required]
        public string Name { get; set; }

        public int CountryId { get; set; }

        public virtual Country Country { get; set; }

        public virtual ICollection<Superhero> Superheroes
        {
            get
            {
                return this.superheroes;
            }
            set
            {
                this.superheroes = value;
            }
        }
    }
}

File: Module III/16.Node.Js/workshop-superheroes/superheroes/server/models/city.js
"use strict";

const mongoose = require("mongoose");

const Schema = mongoose.Schema;

let citySchema = new Schema({
    name: {
        type: String,
        required: true,
        minlength: 2, 
        maxlength: 30,
        unique: true
    },
    country: {
        type: String,
        required: true
    }
});

const City = mongoose.model("city", citySchema);
module.exports = City;